### PR TITLE
fix: stop audio track on mute.

### PR DESCRIPTION
**Pull request by cloudwebrtc**
close https://github.com/livekit/client-sdk-flutter/issues/196.

Tested on iPhone.

### CODE REVIEW

File: lib/src/participant/local.dart [hunk 1]
```diff
@@ -310,13 +313,24 @@ class LocalParticipant extends Participant<LocalTrackPublication> {
...
+        }
       } else {
         if (source == TrackSource.screenShareVideo) {
           await unpublishTrack(publication.sid);
         } else {
           await publication.mute();
+          if (stopOnMute) {
+            await track.sender?.setTrack(null);
+            await track.stop();
+            await track.onTrackStopped();
```
Comment (davidzhao) [inline lib/src/participant/local.dart:332]: Does this work with bluetooth headsets? In browsers, it switches profile from communication to high-quality when the audio track is destroyed.

I think ideally we leave it in the same communication mode (as long as mic track has been acquired). would stopping the track alone make mic recording indicator disappear?

File: lib/src/options.dart [hunk 2]
```diff
@@ -142,9 +142,14 @@ class AudioPublishOptions {
   /// max audio bitrate
   final int audioBitrate;
 
+  /// Turn off the audio track when muted, to avoid the microphone
+  /// indicator light on.
+  final bool stopMicTrackOnMute;
+
   const AudioPublishOptions({
     this.dtx = true,
     this.audioBitrate = AudioPreset.speech,
```
[the remark this answers is not in the document]
Comment (cloudwebrtc) [reply lib/src/options.dart:151]: aha, let me test it.

[the remark this answers is not in the document]
Comment (cloudwebrtc) [reply lib/src/options.dart]: https://github.com/livekit/client-sdk-flutter/pull/207/commits/afe4037d9f1c54ea2e51189fdb3d4ebfcf6eafcf

File: lib/src/participant/local.dart (hunk 1 above)
Comment (cloudwebrtc) [reply lib/src/participant/local.dart:332]: I just tested it with airpods, it doesn't seem to work on the bluetooth headset, I tracked that the mode is set to AVAudioSessionCategoryPlayback, but the mic indicator light will still be on

**Comment by cloudwebrtc** [comment]
hey @janoskranczler, I think this [issue](https://github.com/livekit/client-sdk-flutter/issues/196) is fixed, you can try this branch

File: lib/src/participant/local.dart (hunk 1 above)
Comment (cloudwebrtc) [reply lib/src/participant/local.dart:332]: I have fixed this issue, key steps:
1, set category = playback
2, restart the playout in the audio device module

https://github.com/webrtc-sdk/webrtc/commit/654e4baa96899dd8741d393eafc387653d6ecad9#diff-96494536ec12249ed599120315bbc5df4ca4b7612c4098e5fdce65e72065328dR128-R136

**Comment by janoskranczler** [comment]
@cloudwebrtc Great!
I've tested it on my device and it works! 🎉 
Thank you!

**Review by hiroshihorie — APPROVED** [review]

**Pull request merged**